cranelift/codegen/src/isa/aarch64/lower_inst.rs

```diff
@@ -2397,7 +2397,12 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             // cmp xm, #0
             // cset xm, ne
 
-            let size = VectorSize::from_ty(ctx.input_ty(insn, 0));
+            let s = VectorSize::from_ty(src_ty);
+            let size = if s == VectorSize::Size64x2 {
+                VectorSize::Size32x4
```

Comment:
I'm possibly misunderstanding, but doesn't this only work for the "any" case, not the "all" case, since in the latter case half the lanes will be new artificial zeroes (possibly, if the upper bits of the i64 lanes are zeroes)?

Reply:
Yes, you are absolutely right, but there is a separate arm for `VallTrue` with 64-bit elements on line 2343.

Reply:
Ah, I missed that, thank you! Could you add a comment referring to the case above and noting that `VallTrue` with 64x2 doesn't land here?